codegen/templates/code/controller/event_handlers.gotmpl

```diff
@@ -88,23 +88,23 @@ type generic{{ $resource.Kind }}Handler struct {
     handler {{ $resource.Kind }}EventHandler
 }
 
-func (h generic{{ $resource.Kind }}Handler) Create(object runtime.Object) error {
+func (h generic{{ $resource.Kind }}Handler) Create(object client.Object) error {
```

Comment:
this could be a breaking change for some repos; we should probably mark the changelog as such